Get the correct setup fee limit from SharedPreferences

erdemyerebasmaz opened the pull request:
- Adds "flutter." prefix to SharedPreferences key for setup fee limit on extensions
  - This is the [default prefix](https://github.com/flutter/packages/blob/b21c5424783a9e74cb841a8df7a15f26eba04a7b/packages/shared_preferences/shared_preferences/lib/shared_preferences.dart#L18) used by shared_preferences package

dangeross commented on android/app/src/main/kotlin/com/cBreez/client/BreezForegroundService.kt:
```diff
@@ -39,8 +39,8 @@ class BreezForegroundService : ForegroundService() {
-                "DEBUG" -> Logger.tag(TAG).debug { l.line }
-                "TRACE" -> Logger.tag(TAG).trace { l.line }
+                // "DEBUG" -> Logger.tag(TAG).debug { l.line }
+                // "TRACE" -> Logger.tag(TAG).trace { l.line }
```
Do these need reseting?

erdemyerebasmaz replied:
No, it's continuation of
- #817 
but I'll address it as part of another PR.

erdemyerebasmaz commented:
There's [an issue on iOS](https://stackoverflow.com/a/76355960) where we can't access shared_preferences values from the extension. Addressing this by using [shared preferences with App Group](https://pub.dev/packages/shared_preference_app_group) plugin when setting channel setup fee limit.

dangeross commented on lib/utils/preferences.dart:
```diff
@@ -65,6 +66,8 @@ class Preferences {
     _log.info("set payment options channel setup limit fee : $channelFeeLimitMsat");
     final prefs = await SharedPreferences.getInstance();
     await prefs.setInt(_kPaymentOptionChannelSetupFeeLimit, channelFeeLimitMsat);
+    // iOS Extension requirement
+    await SharedPreferenceAppGroup.setInt(_kPaymentOptionChannelSetupFeeLimit, channelFeeLimitMsat);
```
Is there a need to set it twice? We could also get the preference in the same way using `SharedPreferenceAppGroup.get(_kPaymentOptionChannelSetupFeeLimit)`

dangeross replied:
Makes sense now sorry, it's iOS only

erdemyerebasmaz replied:
Yes, this is workaround for not being able to access `shared_preferences` values on iOS extension. Values set through `shared_preference_app_group` live on app group's UserDefaults and do not overlap with `shared_preferences` values.

dangeross reviewed (approved):
LGTM

erdemyerebasmaz commented:
- Refactored PaymentOptionsPage & PaymentOptionsBloc as there were a lot of unwanted behaviors due to bad state management practices.
- Removed PaymentOptionsBloc tests temporarily.
